packages/core/mesh/network-manager/src/swarm/swarm.ts

```diff
@@ -357,6 +358,7 @@ export class Swarm {
 
     // It is likely that the other peer will also try to connect to us at the same time.
     // If our peerId is higher, we will wait for a bit so that other peer has a chance to connect first.
+    const peer = this._getOrCreatePeer(remoteId);
```

Comment:
Create peer before going to sleep so that peer can be removed if we receive peerLeft event. In this case we won't try to establish a connection after we wake up.